ui/pages/permissions-connect/permissions-connect.container.js

```diff
@@ -37,12 +47,41 @@ const mapStateToProps = (state, ownProps) => {
   const permissionsRequests = getPermissionsRequests(state);
   const currentAddress = getSelectedAddress(state);
 
-  const permissionsRequest = permissionsRequests.find(
+  let permissionsRequest = permissionsRequests.find(
     (req) => req.metadata.id === permissionsRequestId,
   );
 
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  let isUpdatingSnap;
+  if (!permissionsRequest) {
+    permissionsRequest = getSnapUpdateRequests(state).find(
+      (req) => req.id === permissionsRequestId,
+    );
+    if (permissionsRequest) {
```

Comment:
It would be preferable if we could do this in a cleaner way, I'm not entirely sure how though 🤔

Reply:
Is this what you were saying about the shapes of the objects not matching up entirely?

Reply:
Yessir! We can make it clean again by having the new methods I suggested 😄

Reply:
I believe we can fix this by doing something like this: https://github.com/MetaMask/snaps-skunkworks/pull/614

Reply:
Sorry, in this case the shape of the object won't matter because we still need to call `getSnapUpdateRequests` since the `getPermissionsRequests` selector does not check for updates.

Reply:
Actually, I'll write a selector that just grabs both.